### PR TITLE
Add german common crawl dataset

### PR DESCRIPTION
**Pull request by Phil1108**
Adding a subpart of the Common Crawl which was extracted with this repo https://github.com/facebookresearch/cc_net and additionally filtered for duplicates

### CODE REVIEW

File: datasets/german_common_crawl/german_common_crawl.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="data_only", version=VERSION, description="Only the website text without metadata"
+        ),
+        datasets.BuilderConfig(
+            name="first_part",
+            version=VERSION,
+            description="Download only one part (2 GB) instead of everythong (200 GB)",
```
Comment (PhilipMay) [inline datasets/german_common_crawl/german_common_crawl.py:56]: everythong -> everything

File: datasets/german_common_crawl/README.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
+---
+YAML tags:
```
Comment (lhoestq) [inline datasets/german_common_crawl/README.md:2]: We expect to have the YAML tags here, not JSON (even though the tagger app allows to save as JSON)
Can you replace the tags with the YAML instead ?
You can fin the YAML at the top right of the tagger app by clicking `expand yaml tags`

File: datasets/german_common_crawl/README.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+Train only
+
+## Dataset Creation
+
+### Curation Rationale
+
+Handling and Filtering of Common Crawl Data requires large scale Server Ressources at a location in the US (for downloading speed). The total computing time needed to create this dataset is above 100k CPU hours. To give others the opportunity to train models with this dataset easily we make it publicly available. 
+
+In most use cases you see an improved Model Performance when extending the pre-training Data so one can achieve highest accuracies as this is probably the largest available dataset.
```
Comment (lhoestq) [inline datasets/german_common_crawl/README.md:130]:
```suggestion
In most use cases you see an improved Model Performance when extending the pre-training Data so one can achieve highest accuracies as this is probably one of the largest available datasets. 
```
(nit)

File: datasets/german_common_crawl/german_common_crawl.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        if self.config.name == "first_part":
+            files = [dl_manager.download_and_extract(_URL[0])]
+
+        else:
+            files = []
+            for URL in _URL:
+                files.append(dl_manager.download_and_extract(URL))
```
Comment (lhoestq) [inline datasets/german_common_crawl/german_common_crawl.py:112]:
```suggestion
            files = dl_manager.download_and_extract(_URL)
```
this enables parallel downloads

File: datasets/german_common_crawl/german_common_crawl.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+_DESCRIPTION = """\
+German Extract from Common Crawl (High Quality only)
+This Dataset is for pretraining a German Language Model (Unsupervised) or tune a Multilingual Model specifically to German"""
+
+
+_URL = [
+    "https://s3.amazonaws.com/datasets.huggingface.co/datasets/datasets/german-nlp-group/german_common_crawl/de_head_0000_2015-48.tar.gz",
+    "https://s3.amazonaws.com/datasets.huggingface.co/datasets/datasets/german-nlp-group/german_common_crawl/de_head_0000_2016-18.tar.gz",
+]
```
Comment (lhoestq) [inline datasets/german_common_crawl/german_common_crawl.py:41]: Are there only two parts of 2GB ? Not all the 200GB of data ?

File: datasets/german_common_crawl/german_common_crawl.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    def _generate_examples(self, files, split):
+        """ Yields examples. """
+
+        for filepath in files:
+            with open(filepath, "r", encoding="utf-8") as f:
+                for id_, row in enumerate(f):
+                    try:
+                        data = eval(row)
+                    except ValueError:
+                        data = eval(row.replace(chr(0), ""))
```
Comment (lhoestq) [inline datasets/german_common_crawl/german_common_crawl.py:134]: don't use `eval` please

It's not safe at all

**Review by lhoestq — COMMENTED** [review]
Nice thanks !

I left a few comments

File: datasets/german_common_crawl/german_common_crawl.py (hunk 5 above)
Comment (Phil1108) [reply datasets/german_common_crawl/german_common_crawl.py:41]: @thomwolf actually wanted to put this PR on hold until there is a solution for hosting very large datasets at HF. You can see parts our discussions in the datasets-external slack channel

Comment (lhoestq) [reply datasets/german_common_crawl/german_common_crawl.py:41]: Ok right !

**Comment by lhoestq** [comment]
Hi @Phil1108 !
Have you had a chance to take a look at my suggestions ?
Feel free to ping me if you have questions or if you're ready for a review

Thanks again for adding this dataset, this one is very useful !

**Comment by Phil1108** [comment]
> 
> 
> Hi @Phil1108 !
> Have you had a chance to take a look at my suggestions ?
> Feel free to ping me if you have questions or if you're ready for a review
> 
> Thanks again for adding this dataset, this one is very useful !

Hello @lhoestq , 

okay so should we go on with only the 2 parts or wait until the hosting of all 200 GB is cleared at HF? 
If you want to merge it with the 2 parts only at first und upgrade later to all, sure I'll include the minor changes in the next few days and update it here

**Comment by lhoestq** [comment]
I think we can wait a bit to find a solution for the hosting and update the PR to include all the files when it's done.
In the meantime you can update the PR to take into account the suggestions regarding the dataset card, the parallel downloads and the use of eval.

A safer alternative to eval is `ast.literal_eval`, it should do the job

**Comment by albertvillanova** [comment]
Thanks for your contribution, @Phil1108. Are you still interested in adding this dataset?

We are removing the dataset scripts from this GitHub repo and moving them to the Hugging Face Hub: https://huggingface.co/datasets

We would suggest you create this dataset there. Please, feel free to tell us if you need some help.

**Pull request closed without merging**